File: 백준/Silver/11723. 집합/집합.py
import sys

M = int(sys.stdin.readline().strip())
S = set()

for _ in range(M) :
    command = sys.stdin.readline().strip()

    if 'add' in command :
        x = int(command[4:])
        S.add(x)
    elif 'remove' in command :
        x = int(command[7:])
        S.discard(x)
    elif 'check' in command :
        x = int(command[6:])
        if x in S :
            print(1)
        else : print(0)
    elif 'toggle' in command :
        x = int(command[7:])
        if x in S :
            S.remove(x)
        else :
            S.add(x)
    elif command == 'all' :
        S = set(range(1, 21))
    elif command == 'empty' :
        S = set()

File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>

using namespace std;

int S[21];

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int M;
    cin >> M;

    string str;
    int x;
    for(int i = 0; i < M; ++i) {
        cin >> str;

        if(str == "add") {
            cin >> x;
            S[x] = 1;
        }else if(str == "remove") {
            cin >> x;
            S[x] = 0;
        }else if(str == "check") {
            cin >> x;
            if(S[x] == 1) {
                cout << 1 << "\n";
            } else {cout << 0 << "\n";}
        }else if(str == "toggle") {
            cin >> x;
            if(S[x] == 1) {
                S[x] = 0;
            } else {S[x] = 1;}
        }else if(str == "all") {
            for(int i = 1; i <= 20; ++i) {
                S[i] = 1;
            }
        }else {
            for(int i = 1; i <= 20; ++i) {
                S[i] = 0;
            }
        }
    }


    return 0;
}